CMakeLists.txt

```diff
@@ -42,6 +42,21 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Build Halide with ccache if the package is present
+option(Halide_CCACHE_BUILD "Set to ON for a ccache enabled build" OFF)
+if (Halide_CCACHE_BUILD)
+    find_program(CCACHE_PROGRAM ccache)
+    if (CCACHE_PROGRAM)
+        set(Halide_CCACHE_PARAMS CCACHE_CPP2=yes CCACHE_HASHDIR=yes
+            CACHE STRING "Parameters to pass through to ccache")
```

Comment:
Let's add `CCACHE_SLOPPINESS=pch_defines` to this.

Reply:
> Let's add `CCACHE_SLOPPINESS=pch_defines` to this.

According to https://ccache.dev/manual/4.2.html#_precompiled_headers, we have to set it to `pch_defines,time_macros` to get GCC precompiled headers to work. It's not clear what (if anything) `pch_defines` does on its own. The help literally just says "be sloppy". I'd really like to know more before defining it.

Reply:
I just tried both ways and it looks like setting only `pch_defines` and not `time_macros` is faster.

Reply:
I like things that are faster, but I hesitate a bit since (from a quick google) literally every example and everything in the docs show them being set in tandem. We never use the __TIME__ or __DATE__ macros so time_macros should be irrelevant to us, of course.